src/core/mac/sub_mac.cpp

```diff
@@ -357,9 +357,10 @@ void SubMac::StartCsmaBackoff(void)
         if (ShouldHandleTransmitTargetTime())
         {
             if (Time(static_cast<uint32_t>(otPlatRadioGetNow(&GetInstance()))) <
-                Time(mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime) + mTransmitFrame.mInfo.mTxInfo.mTxDelay)
+                Time(mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime) + mTransmitFrame.mInfo.mTxInfo.mTxDelay -
+                    kCcaSampleInterval)
             {
-                mTimer.StartAt(Time(mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime),
+                mTimer.StartAt(Time(mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime) - kCcaSampleInterval,
                                mTransmitFrame.mInfo.mTxInfo.mTxDelay);
             }
             else // Transmit without delay
```

Comment:
BTW, I think that when this `else` is executed, the transmission should be skipped, and error logged, and a new attempt scheduled. Transmitting without delay would most probably be out of RX window of the peer, what is unnecessary channel bandwidth and energy waste.

Reply:
I agree.